src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -227,10 +216,23 @@ internal unsafe QuicConnection(QUIC_HANDLE* handle, QUIC_NEW_CONNECTION_INFO* in
         _localEndPoint = info->LocalAddress->ToIPEndPoint();
     }
 
-    private async ValueTask FinishConnectAsync(QuicClientConnectionOptions options, CancellationToken cancellationToken = default)
+    private static async ValueTask<QuicConnection> StartConnectAsync(QuicClientConnectionOptions options, CancellationToken cancellationToken = default)
```

Comment:
Can we make this local function as it is in the `Socket` example?

Reply:
I can. Is there some interesting difference?

Reply:
It cannot be called outside that function where you define it, whereas private function can be called by any other member of that class. But from the "behavior" perspective, none as far as I can tell.
It's also how I see this pattern implemented in networking code (possible libraries in general, but I cannot speak for that), e.g. socket, httpclient etc.

Reply:
updated.